common/placeholders/src/test/java/io/knotx/server/common/placeholders/JsonPlaceholderSubstitutorTest.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2018 Knot.x Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.server.common.placeholders;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+import io.vertx.core.json.JsonObject;
+
+public class JsonPlaceholderSubstitutorTest {
+
+  private static final JsonObject payload = new JsonObject().put("values",
+      new JsonObject().put("str", "str")
+          .put("number", 1)
+          .put("boolean", true));
+
+  @Test
+  public void shouldReturnValueForString() {
+    //given
+    JsonPlaceholderSubstitutor underTest = new JsonPlaceholderSubstitutor();
+
+    //when
+    String strValue = underTest.getValue(payload, "payload.values.str");
+
+    //then
+    assertEquals("str", strValue);
+  }
+
+  @Test
+  public void shouldReturnValueForNumber() {
+    //given
+    JsonPlaceholderSubstitutor underTest = new JsonPlaceholderSubstitutor();
+
+    //when
+    String numberValue = underTest.getValue(payload, "payload.values.number");
+
+    //then
+    assertEquals("1", numberValue);
+  }
+
+  @Test
+  public void shouldReturnValueForBoolean() {
+    //given
+    JsonPlaceholderSubstitutor underTest = new JsonPlaceholderSubstitutor();
+
+    //when
+    String booleanValue = underTest.getValue(payload, "payload.values.boolean");
+
+    //then
+    assertEquals("true", booleanValue);
```

Comment:
Why not boolean here?

Reply:
as interface method returns String. and placeholder is part of parametrized String...